### DISCUSSION TITLE
option to filter output logs of drawio

### DISCUSSION THREAD

**Discussion by Karreg** [LukeCarrier/mkdocs-drawio-exporter#45] [Ideas]
Hello,

I know the chromium headless dbus dependency is a long running issue, but a quick hack could be to filter related logs.
The warnings pollute the terminal output, and with devcontainer getting more traction, this use case will be more and more frequent.

I have created such an image here: https://gitlab.com/karreg/docker-mkdocs

And I was wondering if it could/should be possible to implement a (long running?) temporary solution to filter out such logs, like this.

Initial:

```
$ xvfb-run -a drawio --no-sandbox --headless --help
[17226:0504/102242.908421:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[17226:0504/102244.208768:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[17226:0504/102244.208877:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[17226:0504/102244.214229:ERROR:bus.cc(399)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[17226:0504/102244.215580:ERROR:bus.cc(399)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
Usage: drawio [options] [input file/folder]

Options:
  -V, --version                                 output the version number
...
```

Hack:

```
$ xvfb-run -a drawio --no-sandbox --headless --help | grep -v "bus.cc"
Usage: drawio [options] [input file/folder]

Options:
  -V, --version                                 output the version number
...
```

Of course the list is not exhaustive. My idea was to not hard code these lines, but make them configurable.

I have tried different things, but I was not able to find a way to do this currently between xvfb-run and drawio options...

What do you think?

Regards.

**Comment by LukeCarrier** [comment]
Hi @Karreg, thanks for reaching out.

I'm against doing this at the moment as filtering the output risks us removing context that might have value.

What we might want to do instead is buffer output from the draw.io sub processes and only print it on a failed export (e.g. non-zero exit status or successful exit without the expected output file being written).

I'm afraid I'm unlikely to do this work in the foreseeable future as I'd prefer to move away from shelling out to the CLI (see #16).

**Comment by Karreg** [comment]
The idea was to propose options to filter out, not filter out by default :)

But I get the other part. I have seen this issue, I was thinking of a more tactical change.

However, for the record, as it might help other people , I have fixed the issue by filtering the `mkdocs` command [this way](https://gitlab.com/karreg/docker-mkdocs/-/blob/324deb28c9819acc5b3ef6be8ca262f9d7ab6bf9/.devcontainer/post_attach.sh#L5).

In case the original repository disappears, here if the command:

```
xvfb-run -a mkdocs serve | grep -i -v "dbus\|gpu\|bus.cc"
```

It can still be disabled in case of error to get the full logs...

Thanks for your work, this is awesome! I need to check if my company can donate...
